### PR TITLE
better log levels

### PR DESCRIPTION
**Pull request by marten-seemann**
Make the log level numbers more intuitive. Logging nothing is now log level 0, debug logging is level 3.

We'll need to change the wiki again ;)

### CODE REVIEW

**Comment by lucas-clemente** [comment]
This is a breaking change, so I'd like to get a bit more reasoning before we do this. I agree that probably our initial choice was weird, but is there a clear advantage of reversing the ordering at this time?

Second, maybe exposing numeric log levels as part of the API are a bad idea in general? What do you think about `QUIC_GO_LOG_LEVEL=DEBUG` instead?

Also, this should be in the changelog.

**Comment by marten-seemann** [comment]
@lucas-clemente: Please have another look.

File: utils/log.go [hunk 1]
```diff
@@ -79,13 +78,12 @@ func init() {
 }
 
 func readLoggingEnv() {
-	env := os.Getenv(logEnv)
-	if env == "" {
-		return
+	switch os.Getenv(logEnv) {
```
Comment (lucas-clemente) [inline utils/log.go]: I'd like to be backwards compatible for the time being (and emit a noticable warning to switch to the new strings). So, please add parsing with the old numeric log levels.

File: utils/log.go [hunk 2]
```diff
@@ -14,13 +13,13 @@ const (
 	logEnv = "QUIC_GO_LOG_LEVEL"
```
Comment (lucas-clemente) [inline utils/log.go]: I'd prefer to have nothing = 0, so move this to a separate const block.

Comment (marten-seemann) [inline utils/log.go]: Fixed.

File: utils/log.go (hunk 1 above)
Comment (marten-seemann) [inline utils/log.go]: Backward compatibility shouldn't be necessary after we merge #648.

**Comment by marten-seemann** [comment]
@lucas-clemente: I just added a Changelog entry. Please approve this PR, then I'll merge it once #648 is merged.

**Comment by lucas-clemente** [comment]
Thanks for adding the line, but I don't think it's particularly helpful like this (what is the change?!).  What about:

```
Changed the log level environment variable to only accept strings ("DEBUG", "INFO", "ERROR"), see [the wiki](https://github.com/lucas-clemente/quic-go/wiki/Logging) for more details.
```

**Comment by lucas-clemente** [comment]
Otherwise LGTM.

**Comment by lucas-clemente** [comment]
Oh, and please update our and caddy's wiki as soon as this is merged, and close #633 :)

**Comment by twdkeule** [comment]
This broke my method to add a prefix to the log 
I will write a PR to add a `QUIC_GO_LOG_PREFIX` ENV if you don't have any objections

**Comment by lucas-clemente** [comment]
Sounds good to me.

**Pull request merged**